qa/tasks/ceph_manager.py

```diff
@@ -1719,25 +1720,10 @@ def do_rados(self, cmd, pool=None, namespace=None, remote=None, **kwargs):
         if remote is None:
             remote = self.controller
 
-        pre = [
-            'adjust-ulimits',
-            'ceph-coverage',
-           f'{self.testdir}/archive/coverage',
-            'rados',
-            '--cluster',
-            self.cluster,
-            ]
-        if pool is not None:
-            pre += ['--pool', pool]
-        if namespace is not None:
-            pre += ['--namespace', namespace]
-        pre.extend(cmd)
-        proc = remote.run(
-            args=pre,
-            wait=True,
-            **kwargs
-            )
-        return proc
+        cmd = (self.rados_cmd + cmd +
+            (['--pool', pool] if pool is not None else []) +
+            (['--namespace', namespace] if namespace is not None else []))
```

Comment:
Oh, okay. I see that now. My intention was to not to use up extra variable (old habit of writing C). Following how it would look otherwise -

```
        opts = []
        if pool is not None:
            opts += ['pool', pool]
        if namespace is not None:
            opts += ['--namespace', namespace]
        cmd = (self.rados_cmd + opts + cmd)
        return remote.run(args=self.rados_cmd + opts + cmd, **kwargs)
```

Reply:
There is no such thing as a redundant variable. The variable avoids this array/tuple building code vomit that is prolific in Python. (Sorry, I feel strongly about this!)

It is much much easier to read:

````
c = [...]
if some_cond:
    c += [more, stuff]
````

than

````
c = [...] + ([more, stuff] if some_cond else [])
````

The logic processing is weird and the line tends to run very long which means more mental exercise to keep track of what's going on (including opening/closing braces). I would strongly urge you to avoid trying to fit as much logic in a single line as possible. It is little more than code golf and doesn't help with maintenance at all.

(Lastly, from a performance perspective, the one-liner version needlessly creates empty lists which adds to memory allocations. This creates more work for Python's already slow garbage collector.)